### PR TITLE
Merge TxBlaster into docker image

### PR DESCRIPTION
**Pull request by ghost**
I had to disable all the volumes. I don't know how they work. Also I guess this image needs to be made properly as you cannot build the image locally on cloud instances. 

Currently the `amount` `address` and `passphrase` are all just put into a file and copied into the image when you run `docker build`, maybe these need to be changed to ENV vars on the `ac` docker container. Really isn't much work to do this if you feel that's a better way to go, but they are fixed variables for all chains either way. 

I did some reading of docker documentation and its fine to put a couple of process inside one container as the TxBlaster is an application that uses 2 executables and a few scripts that link them together.
The mining node is also using the TxBlaster image, you just dont pass it `-blocknotify` on launch and it all it will do is mine.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,9 +8,9 @@
 
 
 ## Build images
-2 images will be built, komodod and marketmaker:
+Build the TxBlaster Image:
 ```
-docker-compose build marketmaker
```
Comment (patchkez) [inline README.md:13]: Please leave building of images via docker-compose there. docker-compose.yml knows how to build/name the image.

File: README.md [hunk 2]
```diff
@@ -22,3 +22,12 @@ docker network create komodo_scale
 ```
 docker-compose up -d
 ```
+
+The seed node will launch and listen for connections on its P2P_PORT, once the ac image has connected to the miner image, the miner will start mining blocks.
```
Comment (patchkez) [inline README.md]: If possible, try to follow proper terminology. Image is built from Dockerfile and Container is started from image.
_once the ac image has connected to the miner image_ - change image->container

File: docker-compose.yml [hunk 3]
```diff
@@ -8,22 +8,22 @@ services:
       args:
         KOMODO_BRANCH: dev
     # image: dragonriders/kmdplatform_komodod_dev:latest
-    image: kmdplatform_komodod_dev
-    restart: on-failure
```
Comment (patchkez) [inline docker-compose.yml]: Why did you remove this line? We definitely need to autorestart container if process crashed.

File: docker-compose.yml [hunk 4]
```diff
@@ -8,22 +8,22 @@ services:
...
+    image: testkomodo
+    #restart: on-failure
     networks:
       - komodo_scale
     ports:
     #  - "32200:32200"
       - "32199:32199"
-    volumes:
```
Comment (patchkez) [inline docker-compose.yml]: Please add all volumes back. What you need to do is to modify .env file and point to some directory on your disk.

File: docker-compose.yml [hunk 5]
```diff
@@ -8,22 +8,22 @@ services:
...
-      - komodo-data:/home/komodo/.komodo
+    #volumes:
+        # - komodo-data:/home/komodo/.komodo
         # - shared-data:/home/komodo/.shared:ro
     environment:
       - ASSET_RPC_USER=kmdrpcus3R
       - ASSET_RPC_PASSWORD=RPcp4ZZw0rD
       - ASSET_RPC_WORKQUEUE=64
-        # - ASSET_BIND=127.0.0.1
-      - ASSET_BIND=0.0.0.0
```
Comment (patchkez) [inline docker-compose.yml:26]: Binding to all addresses was correct. Miner and AC containers will be running on their own private IP addresses. Seed node must listen to 0.0.0.0 so Miner can make connections to it.

File: docker-compose.yml [hunk 6]
```diff
@@ -62,40 +64,19 @@ services:
       -ac_supply=77000000
       -ac_cc=1
       -ac_reward=100000000
```
Comment (patchkez) [inline docker-compose.yml]: End quote is missing?

File: komodod/amount [hunk 7]
```diff
@@ -0,0 +1 @@
+amount=2000
```
Comment (patchkez) [inline komodod/amount]: May I ask you to completely remove this file and pass it as ENV variable? In case we need to change it, we would need to rebuild image.

File: komodod/confd/templates/komodo.conf.tmpl [hunk 8]
```diff
@@ -2,6 +2,7 @@ rpcuser={{ or (getenv "KOMODO_RPC_USER") "rpcuser" }}
 rpcpassword={{ or (getenv "KOMODO_RPC_PASSWORD") "rpcpassword" }}
 server=1
 txindex=1
+blocknotify=/home/komodo/ScaleTestV2/blocknotify.sh %s
```
Comment (patchkez) [inline komodod/confd/templates/komodo.conf.tmpl]: If we keep this line here, both AC and Miner containers will execute this script. If this is not result what we want, we need to pass it as `-blocknotify=...` from docker-compose.yml file.

File: komodod/getblockcount [hunk 9]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+chain=$ASSET_NAME
+komodo-cli -ac_name=$chain getblockcount
```
Comment (patchkez) [inline komodod/getblockcount]: Can we remove all small files and make it as functions inside txblaster script?

File: komodod/importprivkey [hunk 10]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+chain=$ASSET_NAME
+komodo-cli -ac_name=$chain importprivkey Ut1Xx8tAePPxJgTkEVAwj3ZhRvBZ7gY5gJ5ob4QAPiCDcyMJEYgH
```
Comment (patchkez) [inline komodod/importprivkey]: Please pass privkey as ENV variable. Again, we do not want to rebuild images in case this would change.

File: komodod/passphrase [hunk 11]
```diff
@@ -0,0 +1 @@
+passphrase=testpassphraseforsmk702test
```
Comment (patchkez) [inline komodod/passphrase]: Please remove this file and pass it again as ENV.

File: komodod/sendfunds [hunk 12]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+#send transaction and return TXID
+chain=$ASSET_NAME
+source amount
+komodo-cli -ac_name=$chain sendtoaddress RSNWwEWTTFH13LzWVp4EoPscKNxsT5bazt $amount
```
Comment (patchkez) [inline komodod/sendfunds]: You know what to do here :-)

File: komodod/sendfunds [hunk 13]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (patchkez) [inline komodod/sendfunds]: Can this be function of txblaster script?

File: docker-compose.yml (hunk 3 above)
Comment (ghost) [reply docker-compose.yml]: The way the blaster works, if it fails restarting it will do nothing. It will not blast. There is no use having this here. Its better to have it totally fail, and we can then investigate why. If this is a common occurrence in small scale testing, I will need to look at ways of changing how the blast loops are called. I have a few ideas, but if this very simple way works 90%+ of the time I would rather not change it.

File: komodod/confd/templates/komodo.conf.tmpl (hunk 8 above)
Comment (ghost) [reply komodod/confd/templates/komodo.conf.tmpl]: This actually does nothing, it's only used as komodo.conf and didnt actually work for assets. I thought I already removed it lol.

File: docker-compose.yml (hunk 5 above)
Comment (ghost) [reply docker-compose.yml:26]: No the miner is the one who listens. :) This is right.
